File: leetcode/2485-find-the-pivot-integer/constant.go
// Time: O(1)
// Space: O(1)

package main

import "math"

func pivotInteger(n int) int {
	summ := (1 + n) * n / 2
	root := math.Sqrt(float64(summ))
	if float64(int(root)) == root {
		return int(root)
	}
	return -1
}

File: leetcode/2485-find-the-pivot-integer/constant.py
# Time: O(1)
# Space: O(1)


from math import sqrt


class Solution:
    def pivotInteger(self, n: int) -> int:
        summ = (1 + n) * n // 2
        root = sqrt(summ)
        return int(root) if int(root) == root else -1
